harness/features/track.cloud.test.ts

```diff
@@ -0,0 +1,156 @@
+                    .reply((uri, body) => {
+                        mockEvents(body)
+                        return [201, { success: true }]
+                    })
+
+                await wait(1000)
+                await trackResponse.json()
+
+                expect(mockEvents).toHaveBeenCalledTimes(1)
+                const track = mockEvents.mock.calls[0][0]
+                expect(track).toBeDefined()
+
+                expect(track.user.platform).toBe('NodeJS')
+                expect(track.user.sdkType).toBe('server')
+                // expect(track.user.sdkVersion).toBe(latestNodeJsSdkVersion)
+                expect(track.user.user_id).toBe(validUserId)
+
+                expect(track.events.length).toBe(1)
+
+                expect(track.events[0].type).toBe(eventType)
+                expect(track.events[0].target).toBe(variableId)
+                expect(track.events[0].value).toBe(value)
+            })
+
+            it('should retry events API on failed request', async () => {
+
+                const eventType = 'buttonClicked'
+                const variableId = 'json-var'
+                const value = 1
+
+                const response = await createUser(url, { user_id: validUserId })
+                await response.json()
+                const userId = response.headers.get('location')
+
+                const trackResponse = await callTrack(clientId, url, userId,
+                    { type: eventType, target: variableId, value })
+
+                scope
+                    .post(`/client/${clientId}/v1/track`)
+                    .matchHeader('Content-Type', 'application/json')
+                    .reply(519, {})
+
+                scope
+                    .post(`/client/${clientId}/v1/track`)
+                    .matchHeader('Content-Type', 'application/json')
+                    .reply((uri, body) => {
+                        mockEvents(body)
+                        return [201, { success: true }]
+                    })
+
+                await wait(1000)
+                await trackResponse.json()
+
+                expect(mockEvents).toHaveBeenCalledTimes(1)
+                const track = mockEvents.mock.calls[0][0]
+                expect(track).toBeDefined()
+
+                expect(track.user.platform).toBe('NodeJS')
+                expect(track.user.sdkType).toBe('server')
+                // expect(track.user.sdkVersion).toBe(latestNodeJsSdkVersion)
+                expect(track.user.user_id).toBe(validUserId)
+
+                expect(track.events.length).toBe(1)
+
+                expect(track.events[0].type).toBe(eventType)
+                expect(track.events[0].target).toBe(variableId)
+                expect(track.events[0].value).toBe(value)
+
+                expect(scope.isDone()).toBeTruthy()
+            })
+
+        })
+    })
+
+    const callTrack = async (clientId: string, url: string, userLocation: string, event: any) => {
```

Comment:
doesnt matter much here but you should always use `unknown` instead of `any` to indicate "I dont know / care about the type". Using `unknown` prevents you from doing things to the value while not knowing what type it is, where `any` basically disables type checking entirely for that variable